packages/yourloops/components/dialogs/pdf-print-options.tsx

```diff
@@ -0,0 +1,290 @@
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
+
+import { CalendarOrientation } from "../date-pickers/models";
+import RangeDatePicker from "../date-pickers/range-date-picker";
+
+export type Presets = "1week" | "2weeks" | "4weeks" | "3months";
+export interface PrintPDFOptions {
+  /** Print start date (ISO day ex: 2022-02-10) */
+  start: string;
+  /** Print end date (ISO day ex: 2022-02-10) */
+  end: string;
+  preset?: Presets;
+}
+interface DialogPDFOptionsProps {
+  open: boolean;
+  /** Oldest available date date (ISO day ex: 2022-02-10) */
+  minDate: string;
+  /** Newest available date date (ISO day ex: 2022-02-10) */
+  maxDate: string;
+  onResult: (options?: PrintPDFOptions) => void;
+}
+const DEFAULT_PRESET: Presets = "4weeks";
+const MAX_SELECTABLE_DAYS = 90;
+
+const printOptionsStyle = makeStyles((theme: Theme) => {
+  return {
+    marginTop: {
+      marginTop: theme.spacing(2),
+    },
+    customRangeText: {
+      marginBottom: theme.spacing(1),
+    },
+    presetButtons: {
+      width: "20%",
+      marginLeft: theme.spacing(1),
+      [theme.breakpoints.down("sm")]: {
+        width: "40%",
+        marginTop: theme.spacing(1),
+      },
+    },
+    calendarBox: {
+      flexDirection: "column",
+      width: "fit-content",
+      marginLeft: "auto",
+      marginRight: "auto",
+      marginTop: theme.spacing(2),
+      [theme.breakpoints.up("sm")]: {
+        flexDirection: "row",
+      },
+    },
+  };
+}, { name: "dialog-pdf-options" });
+
+function getDatesFromPreset(preset: Presets, minDate: Dayjs, maxDate: Dayjs) {
+  const end = maxDate.format("YYYY-MM-DD");
+  let start: Dayjs;
+  switch (preset) {
+  case "1week":
+    start = maxDate.subtract(6, "days");
+    break;
+  case "2weeks":
+    start = maxDate.subtract(13, "days");
+    break;
+  case "4weeks":
+    start = maxDate.subtract(27, "days");
+    break;
+  case "3months":
+  default:
+    start = maxDate.subtract(89, "days");
+    break;
+  }
+  if (start.isBefore(minDate)) {
+    start = minDate;
+  }
+  return { start: start.format("YYYY-MM-DD"), end, preset };
+}
+
+function DialogPDFOptions(props: DialogPDFOptionsProps) {
+  const { open, onResult } = props;
+  const { t } = useTranslation("yourloops");
+  const theme = useTheme();
+  const matchLandscape = useMediaQuery(theme.breakpoints.up("sm"));
+  const fullScreen = useMediaQuery(theme.breakpoints.down("sm"));
+  const classes = printOptionsStyle();
+  const orientation: CalendarOrientation = matchLandscape ? "landscape" : "portrait";
+
+  const [customStartDate, setCustomStartDate] = React.useState<Dayjs|null>(null);
+  const { minDate, maxDate } = React.useMemo(() => {
+    let minDate = dayjs(props.minDate, { utc: true });
```

Comment:
Sonar is giving me a warning about `minDate` and `maxDate` being already defined in the upper scope (line 133)

Reply:
Good catch, forgot to add eslint rule: "no-shadow" :cry: